File: frontend/src/app/services/students.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { BASE_URL } from '../constants';

@Injectable({
  providedIn: 'root'
})
export class StudentsService {

  constructor(private httpClient : HttpClient) { }

  public findAll() : Observable<any> {
    return this.httpClient.get( BASE_URL +'/students/');
  }

  public save(student) : Observable<any> {
    return this.httpClient.post( BASE_URL +'/students/', student);
  }

  public remove(id : String) : Observable<any> {
    return this.httpClient.delete( BASE_URL +'/students/'+id);
  }

  public findByParentId(parentId : String) : Observable<any>  {
    return this.httpClient.get( BASE_URL +'/students/parent/'+parentId);
  }

  public findOneById(id : String) : Observable<any>  {
    return this.httpClient.get( BASE_URL +'/students/'+id);
  }

  public findOneByUsername(username : String) : Observable<any>  {
    return this.httpClient.get( BASE_URL +'/students/username/'+username);
  }
}

File: backend/src/main/java/com/zka/lyceena/services/StudentsService.java
package com.zka.lyceena.services;

import com.zka.lyceena.dto.StudentDto;
import com.zka.lyceena.entities.actors.Student;

import java.util.List;

public interface StudentsService {

    List<Student> findAll();
    void save(StudentDto dto);
    void deleteById(String id);
    List<Student> findByParentId(String parentId);
    Student findOne(String id);
    Student findOneByUsername(String username);
}
